src/Native/Unix/System.Native/pal_interfaceaddresses.cpp

```diff
@@ -44,6 +44,12 @@ extern "C" int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onI
     for (ifaddrs* current = headAddr; current != nullptr; current = current->ifa_next)
     {
         uint32_t interfaceIndex = if_nametoindex(current->ifa_name);
+        // ifa_name may be an aliased interface name.
+        // Use if_indextoname to map back to the true device name.
+        char actualName[IF_NAMESIZE];
+        char* result = if_indextoname(interfaceIndex, actualName);
+        assert(result == actualName && result != nullptr);
```

Comment:
Would this be better as a real error result check rather than as a debug-only assert?  What if if_indextoname fails?  From the man page, it looks like it could, if only in a very rare case.  For example, isn't it possible there's a race condition where something else ends up changing system settings/configuration/etc. between the time the index was retrieved and when we try to get its name?